.github/draft_release_notes.sh

```diff
@@ -0,0 +1,54 @@
+#! /bin/sh
+
+# this script assumes the GITHUB_TOKEN and PREVIOUS_TAG environment variables have been set;
+# it produces a 'Changes.md' file as its final output;
+# the file 'last-300-prs-with-release-note.txt' that is produces is intermediate data; it is not
+# pruned for now to assist development of the release notes process (we are still curating all this)
+
+sudo apt-get -y install jq
+sudo apt-get -y install wget
+curl -L https://github.com/github/hub/releases/download/v2.14.2/hub-linux-amd64-2.14.2.tgz | tar xzf -
+PWD=`pwd`
+export PATH=$PWD/hub-linux-amd64-2.14.2/bin:$PATH
+git fetch --all --tags --prune --force
+echo -e "# Insert Title\n" > Changes.md
+echo -e "## Features\n\n## Fixes\n\n## Backwards incompatible changes\n\n## Docs\n\n## Misc\n\n## Thanks" >> Changes.md    - name: Draft Release
+# this effectively gets the commit associated with github.event.inputs.tags
+COMMON_ANCESTOR=$(git merge-base $PREVIOUS_TAG HEAD)
+# in theory the new tag has not been created yet; do we want another input that specifies the existing
+# commit desired for drafting the release? for now, we are using HEAD in the above git merge-base call
+# and PR cross referencing below
+
+# use of 'hub', which is an extension of the 'git' CLI, allows for pulling of PRs, though we can't search based on commits
```

Comment:
Should we use `hub` or the official [GitHub CLI](https://cli.github.com/) for this? So far I am quite happy with `gh` when it comes to command line work against GitHub.

Reply:
the tekton work I borrowed from used `hub` @HeavyWombat 

I'm unfamiliar with `gh`.  I'll have to spend some cycles to see if it has equivalent functionality (unless you already have some code suggestions on how to install that instead and similarly query PRs )

Otherwise, I'll report back when I have some info.

Reply:
So yeah it supports the same list verb and arguments:  https://cli.github.com/manual/gh_pr_list

install for ubuntu:  https://github.com/cli/cli/blob/trunk/docs/install_linux.md#debian-ubuntu-linux-apt

based on research, the net is gh appears to be hub's successor .... see issue 312 from https://github.com/cli/cli as well as https://mislav.net/2020/01/github-cli/

installing on my fedora box and confirming it works for me

Reply:
spoke too soon ... missed that `gh pr list` does not support the `-f` options that `hub` provides

that actually is critical to the line parsing that is subsequently done

so I'm going to have to stick to `hub` for now @HeavyWombat 
but I'll add a comment that if `gh` add `-f` support we can switch to that

Reply:
```
unknown shorthand flag: 'f' in -f

Usage:  gh pr list [flags]

Flags:
  -a, --assignee string   Filter by assignee
  -B, --base string       Filter by base branch
  -l, --label strings     Filter by labels
  -L, --limit int         Maximum number of items to fetch (default 30)
  -s, --state string      Filter by state: {open|closed|merged|all} (default "open")
  -w, --web               Open the browser to list the pull requests
```

Reply:
Thanks for the investigations! I did not know it lacks that formatting feature.